### PR TITLE
[RPD-211] [BUG] Fix matcha get with updated state file

### PR DESCRIPTION
**Pull request by Christopher-Norman**
Currently, `matcha get` throws an error with the new state file. This is likely due to the added fields:

location": "ukwest",
"prefix": "xxxx"
As these are not dictionaries an error is thrown when matcha get is called

We have added these values in the correct format to the Terraform outputs file to fix this.

## Checklist

Please ensure you have done the following:

* [x] I have read the [CONTRIBUTING](https://github.com/fuzzylabs/matcha/blob/main/CONTRIBUTING.md) guide.
* [x] I have updated the documentation if required.
* [x] I have added tests which cover my changes.

## Type of change

Tick all those that apply:

* [x] Bug Fix (non-breaking change, fixing an issue)
* [ ] New feature (non-breaking change to add functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Other (add details above)

### CODE REVIEW

File: src/matcha_ml/infrastructure/output.tf [hunk 1]
```diff
@@ -56,5 +56,16 @@ output "model_deployer_seldon_base_url" {
 }
 
 output "cloud_azure_resource_group_name" {
+  description = "The full name of the resource group"
   value = module.resource_group.name
 }
+
+output "cloud_azure_prefix" {
```
Comment (dudeperf3ct) [inline src/matcha_ml/infrastructure/output.tf:63]: I don't think these are required, as we are already getting and updating these values on line 185 of `azure_template.py`

File: src/matcha_ml/infrastructure/output.tf [hunk 2]
```diff
@@ -56,5 +56,16 @@ output "model_deployer_seldon_base_url" {
...
+  description = "The prefix for the resource group"
+  value = var.prefix
+}
+
+output "cloud_azure_location" {
```
Comment (dudeperf3ct) [inline src/matcha_ml/infrastructure/output.tf:68]: Same here

File: src/matcha_ml/templates/run_template.py [hunk 3]
```diff
@@ -205,11 +205,8 @@ def _update_state_file(self, state_outputs: Dict[str, Dict[str, str]]) -> None:
         Args:
             state_outputs (Dict[str, Dict[str, str]]): Dictionary containing outputs to be written to state file.
         """
-        with open(self.state_file) as fp:
-            out_data = json.load(fp)
-        out_data.update(state_outputs)
         with open(self.state_file, "w") as fp:
```
Comment (dudeperf3ct) [inline src/matcha_ml/templates/run_template.py:208]: This would be read, update and write.

**Review by dudeperf3ct — APPROVED** [review]
Good work @Christopher-Norman. LGTM!

**Review by dudeperf3ct — APPROVED** [review]

**Review by d-lowl — APPROVED** [review]
LGTM

**Comment by d-lowl** [comment]
Closing, since it's been implemented in a different PR as a hotfix on a stable release #118

**Pull request closed without merging**